packages/admin/src/FilamentManager.php

```diff
@@ -33,18 +34,37 @@ class FilamentManager
 
     protected array $widgets = [];
 
+    protected ?Closure $navigationBuilder = null;
+
     public function auth(): Guard
     {
         return auth()->guard(config('filament.auth.guard'));
     }
 
+    public function navigation(Closure $builder): void
+    {
+        $this->navigationBuilder = $builder;
+    }
+
+    public function buildNavigation(): array
+    {
+        /** @var \Filament\Navigation\NavigationBuilder $builder */
+        $builder = app()->call($this->navigationBuilder, [
```

Comment:
We could remove the `app()->call()` params and allow the container to instantiate a `NavigationBuilder`, right?

Reply:
Yeah we could do actually. There's no dependencies so it'll just make a new instance anyway.

Reply:
Great